packages/hub/services/scheduled-payments/fetcher.ts

```diff
@@ -17,30 +17,41 @@ export default class ScheduledPaymentsFetcherService {
   // It is also important to note that validForDays tells us for how many days the payment is still valid to retry in
   // case it is failing for some reason.
 
+  // In the case of scheduled payments that failed and need to be retried, the query will return only the scheduled payments
+  // where enough time has passed since the last failed payment attempt. We allow maximum 8 failed attempts, and the time
+  // between attempts is hardcoded in the query (5 minutes, 1 hour, 5 hours, 12 hours, 12 hours, 12 hours, 12 hours, 12 hours)
   async fetchScheduledPayments(limit = 10): Promise<ScheduledPayment[]> {
     let prisma = await this.prismaManager.getClient();
-    let _nowUtc = nowUtc();
+    let now = this.clock.utcNow();
+    let nowString = now.toISOString(); // We use ISO string to make sure we operate on UTC dates only. Otherwise Prisma will use local time zone.
 
     let results: [{ id: string; started_at: Date }] = await prisma.$queryRaw`SELECT 
         scheduled_payments.id AS id,
         last_failed_payment_attempt.started_at
       FROM 
         scheduled_payments
-      LEFT JOIN (SELECT scheduled_payment_attempts.scheduled_payment_id,
-                  MAX(scheduled_payment_attempts.started_at) AS started_at
-            FROM scheduled_payment_attempts
-            WHERE scheduled_payment_attempts.status = 'failed'
-            GROUP BY scheduled_payment_id) AS last_failed_payment_attempt
+      LEFT JOIN (
+        SELECT
+          scheduled_payment_attempts.scheduled_payment_id,
+          MAX(scheduled_payment_attempts.started_at) AS started_at,
+          COUNT(*) as failed_attempts_count
+        FROM scheduled_payment_attempts
+        INNER JOIN scheduled_payments ON scheduled_payments.id = scheduled_payment_attempts.scheduled_payment_id
+        WHERE scheduled_payment_attempts.status = 'failed'
+          AND started_at >= scheduled_payments.pay_at
```

Comment:
Only count failed attempts in the most recent schedule period (when the payment is recurring)